[Workspace] Add documentation for workspace and ACL (#8643)
opensearch-project/documentation-website: _dashboards/workspace/workspace.md

---
layout: default
title: Workspace for OpenSearch Dashboards
nav_order: 110
has_children: true
---

# Workspace for OpenSearch Dashboards
Introduced 2.18
{: .label .label-purple }

> The Workspace feature in OpenSearch Dashboards enables you to tailor your environment with use-case-specific configurations. For example, you can create dedicated workspaces for observability scenarios, allowing you to focus on relevant functionalities. Additionally, the Workspace feature enables organization of visual assets, such as dashboards and visualizations, within a workspace that isolated storage.

Reviewer: ```suggestion
The Workspace feature in OpenSearch Dashboards enables you to tailor your environment with use-case-specific configurations. For example, you can create dedicated workspaces for observability scenarios, allowing you to focus on relevant functionalities. Additionally, the Workspace feature enables organization of visual assets, such as dashboards and visualizations, within a workspace with isolated storage.
```

The following saved object shows a dashboard object associated with the workspace by `M5NqCu`:

```typescript
{
  type: "dashboard",
  id: "da123f20-6680-11ee-93fa-df944ec23359",
  workspaces: ["M5NqCu"]
}
```
{% include copy-curl.html %}

Saved objects support association with multiple workspaces, facilitating cross-team collaboration and resource sharing. This feature is useful when an object is relevant to various team, projects, or use cases. 

The following example shows a data source object linked to multiple workspaces:

```typescript
{
  type: "data-source",
  id: "da123f20-6680-11ee-93fa-df944ec23359",
  workspaces: ["M5NqCu", "<TeamA-workspace-id>", "<Analytics-workspace-id>"]
}
```
{% include copy-curl.html %}

## Non-workspace saved objects

Not all saved objects in OpenSearch Dashboards are workspace-associated. Some objects operate independently of workspace framework. These objects lack `workspace` attribute and serve system-wide functions. For example, the global user interface settings object manages configurations affecting the entire OpenSearch Dashboards interface to maintain consistent functionality across all workspaces.

> This dual approach, workspace-associated and independent objects, allows OpenSearch Dashboards to balance granular, context-specific customization with overall system consistency. 

Reviewer 2: ```suggestion
This dual approach allows OpenSearch Dashboards to balance granular, context-specific customization with overall system consistency. 
```

## Enabling the Workspace feature

In your `opensearch_dashboards.yml` file, set the following option:

```yaml
workspace.enabled: true
uiSettings:
  overrides:
    "home:useNewHomePage": true
```
{% include copy-curl.html %}

> If your cluster has the security plugin installed, then multitenancy must be disabled to avoid conflicts with workspaces of similar functionality.

Reviewer 2: ```suggestion
If your cluster has the Security plugin installed, then multi-tenancy must be disabled to avoid conflicts with workspaces providing similar functionality:
```

> 

Reviewer 2: Above: "workspaces providing similar functionality" => "similar workspaces"?

---
layout: default

> title: Workspace

Reviewer: ```suggestion
title: Workspace for OpenSearch Dashboards
```

nav_order: 110
has_children: true
---

# Workspace for OpenSearch Dashboards

> **Introduced 2.17**

Reviewer: ```suggestion
Introduced 2.18
{: .label .label-purple }
```

---
layout: default
title: Workspace
nav_order: 110
has_children: true
---

# Workspace for OpenSearch Dashboards
Introduced 2.18
{: .label .label-purple }

> {: .label .label-purple }

Reviewer: ```suggestion
```
